jax/api.py

```diff
@@ -354,12 +482,72 @@ def xla_computation(fun: Callable,
     ROOT tuple.18 = (f32[], f32[], f32[]) tuple(all-reduce.7, all-reduce.12, all-reduce.17)
   }
   """
+  internal_computation_maker = _xla_computation(
+      fun,
+      static_argnums=static_argnums,
+      axis_env=axis_env,
+      in_parts=in_parts,
+      out_parts=out_parts,
+      backend=backend,
+      tuple_args=tuple_args,
+      instantiate_const_outputs=instantiate_const_outputs)
+
+  def computation_maker(*args, **kwargs):
+    xla_return = internal_computation_maker(*args, **kwargs)
+    if return_shape:
+      return xla_return.xla_computation, xla_return.out_shape
+    else:
+      return xla_return.xla_computation
+
+  return computation_maker
+
+
+class _XlaReturn(NamedTuple):
+  xla_computation: xc.XlaComputation
+  # A tree of `ShapeDtypeStruct`.
+  out_shape: Any
+  out_pytree_def: Any  # A `PyTreeDef` object.
+  lazy_expressions: Optional[List[xla.lazy.LazyExpr]]
+  shaped_arrays: List[core.ShapedArray]
+  parameter_is_tupled_arguments: bool
+
+
+def _xla_computation(
+    fun: Callable,
+    static_argnums: Union[int, Iterable[int]] = (),
+    axis_env: Optional[Sequence[Tuple[AxisName, int]]] = None,
+    in_parts=None,
+    out_parts=None,
+    backend: Optional[str] = None,
+    tuple_args: Optional[bool] = None,
+    instantiate_const_outputs: Optional[bool] = True,
+    return_shape: bool = False,
+    xla_return: bool = False,
+    donate_argnums: Union[int, Iterable[int]] = ()
+) -> Callable:
+  """An internal implementation for `xla_computation` and `_cpp_jit`.
+
+  See `xla_computation` for the full documentation.
+
+  Args:
+    tuple_args: By default (None), tupling will be enabled when there are more
+      than 100 arguments, since some platforms have limits on argument arity.
+
+  Returns:
+    An `_XlaReturn` object, some field possibly being None depending on
```

Comment:
The comment also seems stale?

Reply:
Yes. I was waiting for the anwser to the other one, to know which action to take, before updating everything.

it's also because I'm in Google3, on Fig, so each time to make my changes visible, I need to Amend + Upload + run the export script. So I had some changes but not all.